apps/www/src/components/share/share-overlay.tsx

```diff
@@ -2,16 +2,131 @@
 import { css } from '@app/styled-system/css'
 import * as Dialog from '@radix-ui/react-dialog'
 import { Root as VisuallyHidden } from '@radix-ui/react-visually-hidden'
-import { IconClose } from '@republik/icons'
+import {
+  IconClose,
+  IconLink,
+  IconLogoFacebook,
+  IconLogoTelegram,
+  IconLogoThreema,
+  IconLogoWhatsApp,
+  IconMail,
+} from '@republik/icons'
+import { ComponentPropsWithoutRef, useState } from 'react'
+import { ShareProps } from './types'
+import copyToClipboard from 'clipboard-copy'
 
-export function ShareOverlay(props: {
-  triggerLabel: React.ReactNode
-  children: React.ReactNode
+function ShareButton({
+  icon: Icon,
+  label,
+  href,
+  onClick,
+}: {
+  icon: React.ComponentType<ComponentPropsWithoutRef<typeof IconLogoFacebook>>
+  label: string
+  href?: string
+  onClick?: (e: React.MouseEvent<HTMLAnchorElement>) => void
 }) {
+  return (
+    <a
+      className={css({
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        flex: '0 1 auto',
+        cursor: 'pointer',
+        margin: '4',
+        gap: '2',
+        color: 'text',
+        textDecoration: 'none',
+      })}
+      target='_blank'
+      href={href}
+      onClick={onClick}
+      rel='noopener noreferrer'
+    >
+      <Icon size={24} />
+      <span>{label}</span>
+    </a>
+  )
+}
+
+export function ShareOverlay({
+  triggerLabel,
+  title,
+  url,
+  emailSubject,
+  emailBody,
+  emailAttachURL = true,
+}: {
+  triggerLabel: React.ReactNode
+  emailBody?: string
+  emailAttachURL?: boolean
+} & ShareProps) {
+  const [linkCopied, setLinkCopied] = useState(false)
+
+  const shareOptions = [
+    {
+      name: 'facebook',
+      target: '_blank',
+      href: `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(
+        url,
+      )}`,
+      icon: IconLogoFacebook,
+      title: 'Auf Facebook teilen',
+      label: 'Facebook',
+    },
+    {
+      name: 'whatsapp',
+      target: '_blank',
+      href: `https://api.whatsapp.com/send?text=${encodeURIComponent(url)}`,
+      icon: IconLogoWhatsApp,
+      title: 'Auf WhatsApp teilen',
+      label: 'WhatsApp',
+    },
+    {
+      name: 'threema',
+      target: '_blank',
+      href: `https://threema.id/compose?text=${encodeURIComponent(url)}`,
+      icon: IconLogoThreema,
+      title: 'Auf Threema teilen',
+      label: 'Threema',
+    },
+    {
+      name: 'telegram',
+      target: '_blank',
+      href: `https://t.me/share/url?url=${encodeURIComponent(url)}`,
+      icon: IconLogoTelegram,
+      title: 'Auf Telegram teilen',
+      label: 'Telegram',
+    },
+    {
+      name: 'mail',
+      href: `mailto:?subject=${encodeURIComponent(
+        emailSubject,
+      )}&body=${encodeURIComponent(
+        emailBody + (emailAttachURL ? '\n\n' + url : ''),
```

Comment:
I think I would just do `encodeURIComponent(url)` here. I don't think emailBody and emailAttachURL are useful.